src/components/Dropdown/Items/SingleSelectItem/SingleSelectItem.tsx

```diff
@@ -1,45 +1,71 @@
 
-export interface Props extends Omit<DropdownMenuItemProps, 'onSelect'> {
+export interface Props<PriorityItemValue> extends Omit<DropdownMenuItemProps, 'onSelect'> {
     /** A function to be called if the item is clicked */
     onSelect?: (e: React.SyntheticEvent<HTMLDivElement>) => void;
     /** Id for the checkbox */
     id?: string;
-    /** to add a priority badge before the label */
-    hasPriority?: boolean;
     /** Checkbox status */
     isSelected?: boolean;
+    /** Should PrioritySelector be included */
+    hasPriority?: boolean;
+    /** props for PrioritySelector */
+    priority?: PrioritySelectorProps<PriorityItemValue>;
 }
 
 const { block } = bem('DropdownItem', styles);
 
-export const SingleSelectItem = React.forwardRef<HTMLDivElement, Props>(
+export const SingleSelectItem = React.forwardRef<HTMLDivElement, Props<string>>(
     (
-        {
-            children,
-            isSelected = false,
-            disabled = false,
-            hasPriority = false, // eslint-disable-line @typescript-eslint/no-unused-vars
-            ...rest
-        },
+        { children, isSelected = false, disabled = false, hasPriority = false, priority, ...rest },
         ref
-    ) => (
-        <DropdownMenuItem
-            ref={ref}
-            role="option"
-            aria-selected={isSelected}
-            tabIndex={disabled ? -1 : 0}
-            {...rest}
-            {...block({
-                isSelected,
-                disabled,
-                ...rest,
-            })}
-        >
-            {children}
-        </DropdownMenuItem>
-    )
+    ) => {
+        const hasPriorityList = hasPriority && priority && priority.list.length > 0;
+        const priorityRef = React.useRef<HTMLDivElement>(null);
+
+        const stopPropagation = (e: React.MouseEvent | React.KeyboardEvent) => {
+            e.stopPropagation(); // Prevents the menu from closing
+        };
+
+        const handleKeyDown = (e: React.KeyboardEvent) => {
+            if (e.key === 'Tab') {
+                if (priorityRef.current) {
+                    e.preventDefault();
+                    priorityRef.current.focus();
+                }
+            }
+        };
+
+        return (
+            <DropdownMenuItem
+                ref={ref}
+                role="option"
+                aria-selected={isSelected}
+                tabIndex={disabled ? -1 : 0}
+                onKeyDown={handleKeyDown}
+                {...rest}
+                {...block({
+                    isSelected,
+                    disabled,
+                    ...rest,
+                })}
+            >
+                {hasPriorityList && (
+                    <div role="none" onClick={stopPropagation} onKeyDown={stopPropagation}>
+                        <PrioritySelector
+                            list={priority.list}
+                            selectedItem={priority.selectedItem}
+                            onSelect={priority.onSelect}
```

Comment:
shouldn't other props e.g. `isDisabled` and so on be passed here too? Maybe easiest to do `<PrioritySelector {...priority} buttonRef={priorityRef} />`

Reply:
same same... already did it before... doing it again now!